salt/states/file.py

```diff
@@ -607,29 +608,34 @@ def _check_file(name):
     return ret, msg
 
 
-def _clean_dir(root, keep, exclude_pat):
+def _find_keep_files(root, keep):
     '''
-    Clean out all of the files and directories in a directory (root) while
-    preserving the files in a list (keep) and part of exclude_pat
+    Compile a list of valid keep files (and directories).
     '''
-    removed = set()
     real_keep = set()
     real_keep.add(root)
-    if isinstance(keep, list):
+    if isinstance(keep, collections.Iterable):
```

Comment:
This would make this code iterate over strings:

```python
>>> import collections
>>> isinstance('foo', collections.Iterable)
True
```

Reply:
Yes, I guess the question is why would we get a string here. My intent was to not require a list excluding some other iterator or generator.

Reply:
Since this is a bugfix and this change is not essential to the fix, I've reverted it.

Reply:
This is true, I think I was more concerned with changing from list to something that would be more inclusive to types as a potential source of future bugs, and didn't look at the greater context.

Is there a reason to think that other iterable types could be used here? At the moment it does not appear that is the case, but if so, I think this would be a good change to make.